harness/determined/pytorch/_pytorch_context.py

```diff
@@ -57,4 +82,224 @@ def get_lr_scheduler(self) -> Optional[pytorch.LRScheduler]:
             * ``optimizer()``
             * ``create_lr_scheduler()``
         """
-        return self.lr_scheduler
+        # TODO(DET-3267): deprecate this when releasing pytorch flexible primitives.
+        check.lt_eq(len(self.lr_schedulers), 1)
+        if len(self.lr_schedulers) == 1:
+            return self.lr_schedulers[0]
+        return None
+
+    def _load_trial(self, trial: "pytorch.PyTorchTrial") -> None:
+        # TODO(DET-3262): remove this backward compatibility of old interface.
+        if (
+            util.is_overridden(trial.build_model, pytorch.PyTorchTrial)
+            or util.is_overridden(trial.optimizer, pytorch.PyTorchTrial)
+            or util.is_overridden(trial.create_lr_scheduler, pytorch.PyTorchTrial)
+            or util.is_overridden(trial.train_batch, pytorch.PyTorchTrial)
+        ):
+            check.true(
+                util.is_overridden(trial.build_model, pytorch.PyTorchTrial)
+                and util.is_overridden(trial.optimizer, pytorch.PyTorchTrial)
+                and util.is_overridden(trial.train_batch, pytorch.PyTorchTrial),
+                "Must define build_model(), optimizer(), and train_batch().",
+            )
+
+            model = self._Model(trial.build_model())
+            optim = self._Optimizer(trial.optimizer(model))
+
+            lr_scheduler = trial.create_lr_scheduler(optim)
+            if lr_scheduler is not None:
+                self._LRScheduler(lr_scheduler.scheduler, lr_scheduler.step_mode)
+
+            amp_setting = (
```

Comment:
blocking: use `self._use_amp()`

Reply:
This is for backward compatibility of configuring AMP in the experiment configuration. `self._use_amp()` is used after configuring AMP.

Reply:
Should use `ExperimentConfig.mixed_precision_enabled()` instead here then. Also should avoid ordering requirements between functions when it's not necessary to have them.

Reply:
> Since we are moving from configuring mixed precision in the experiment config to configure_amp, we will remove mixed_precision_enabled function at the same time. So it is not a good idea to rely on the code of the old interface.

Right, but you would be using it here only to check whether it's configured via the old interface. So once that is deprecated you won't need this check anymore.

Reply:
DOne.